code/datums/gamemode/factions/bloodcult/bloodcult_buildings.dm

```diff
@@ -1068,13 +1066,10 @@ var/list/cult_spires = list()
 	for (var/obj/O in loc)
 		if (O != src)
 			O.ex_act(2)
+		if(!O.gcDestroyed && (istype(O, /obj/structure) || istype(O, /obj/machinery)))
+			qdel(O)
```

Comment:
Hold on, this will totally cause the pillar to destroy itself, won't it? you're not checking for O != src